base/operators.jl

```diff
@@ -1211,6 +1211,19 @@ used to implement specialized methods.
 """
 <(x) = Fix2(<, x)
 
+"""
+    hasproperty(name::Symbol)
+
+Create a function that indicates whether its argument has the property `name` using
+[`hasproperty(x, name)`](@ref), i.e. a function equivalent to `x -> hasproperty(x, name)`.
+The returned function is of type `Base.Fix2{typeof(hasproperty)}`, which can be
+used to implement specialized methods.
+
+!!! compat "Julia 1.9"
+    This functionality requires at least Julia 1.9.
+"""
+hasproperty(s::Union{Symbol,String,Int}) = Fix2(hasproperty, s)
```

Comment:
```suggestion
hasproperty(f::Symbol) = x -> hasproperty(x, f)
```

Reply:
is the type restriction necessary for this to work?

Reply:
It allows us to generate more efficient code when the type of field name can't be inferred.

Reply:
Ah, but I just forgot the use case in DataFrames.jl. I think we can remove the type restriction of `f` if that is still the case.

Reply:
It would be good to support both but I'm not sure if there are pros and cons to using the fixed method for setting like strings. Can we just lump them all in one anonymous method without issue